### ISSUE TITLE
FormattedHTMLMessage showing blank w/error on version 3.0.0-beta.14 on React >16.8

### ISSUE THREAD

**Issue by lukesmmr** [formatjs/react-intl#1360]
Albeit usage is discouraged I'm using `FormattedHTMLMessage` for some segments introducing span classes for styling purposes. 

### Expected behavior

Correct output of `FormattedHTMLMessage` in React Components with the following syntax:

```
<FormattedHTMLMessage
    id="segment.span_class"
    defaultMessage={
         'Some <span class="cssclass">message</span> wrapped in a span class'
    }
/>
```

### Current behavior

After updating to the latest beta `3.0.0-beta.14` (from `^2.9.0`) any segment using `FormattedHTMLMessage` is showing blank and throwing a console error:

```
Warning: Invalid prop `dangerouslySetInnerHTML` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.
    in Fragment (created by FormattedHTMLMessage)
```

### Step to reproduce for BUG REPORT

Install `3.0.0-beta.14` in your project on React `>16.8` and attempt usage of `FormattedHTMLMessage`. 
Reverting to `^2.9.0` instantly solves the issue and the segments are displayed again. 

### Your Environment

|          Executable | Version |
| ------------------: | :------ |
| `npm ls react-intl` | 3.0.0-beta.14 |
|      `npm ls react` | ^16.8.6 |
|     `npm --version` | 6.1.0 |
|    `node --version` | v10.3.0 |

| OS   | Version |
| ---- | ------- |
| MacOs High Sierra | 10.13.6 |

| Browser | Version |
| ------- | ------- |
| Chrome | 75.0.3770.100 |

**Comment by longlho** [comment]
We recently switched default `textComponent` to `<>` which is why this is an issue. You can manually override by setting `tagName="span"`. I'll update the Upgrade Guide to account for this. Thanks for reporting!

**Comment by DragonRaider5** [comment]
Shall we maybe keep `<span>` as the default for `FormattedHTMLMessage`? Requiring the user to set a prop for it to work at all doesn't seem right...

**Comment by longlho** [comment]
yeah that works as well. Although the behavior will still slightly change:
Before it was `if tagName is nothing, use textComponent`, now it'll be `if tagName is nothing, use defaultProps.tagName, then textComponent`

**Comment by longlho** [comment]
fixed in #1361

**Comment by lukesmmr** [comment]
@longlho Awesome, thanks for the fast response and fix. 💯
